service/controller/resource/app/desired.go

```diff
@@ -188,6 +191,44 @@ func (r *Resource) newApp(appOperatorVersion string, cr apiv1alpha2.Cluster, app
 	}
 }
 
+func chartName(appName, catalog, version string) (string, error) {
+	// first fetch
+	chartName := appName
+	catalogBaseURL := fmt.Sprintf("https://giantswarm.github.io/%s-catalog", catalog)
```

Comment:
The catalog naming is sometimes inconsistent. If this is just intended for the default and default-test catalogs it will work but I'm not keen on the hardcoded URLs here.

We added to apptest a getCatalogURL function and a list of the known catalogs. What if we moved this to the appcatalog library?

https://github.com/giantswarm/apptest/blob/master/apptest.go#L687-L689

Reply:
This would be nice. I'll make the required changes and then update this PR

Reply:
Since I want to get this out of the door, WDYT about leaving this as is for now?

I checked the implementation in `apptest` you linked. As I see it, `apptest` defines its own `App` struct, which would need to also move into `appcatalog`. But isn't it already defined in apiextensions somewhere? 

I've also seen a nice way to get the catalog url in config-controller here https://github.com/giantswarm/config-controller/blob/0dc09d6ce09b3d35baf552a6c618aae1b1613e34/service/controller/handler/configversion/create.go#L119-L147

But: I've no idea how to replace the used `CtrlClient` there to make it work here. I'm not experienced in operator programming..

Reply:
> Since I want to get this out of the door, WDYT about leaving this as is for now?

@ubergesundheit Yes sure let's go with this for now.

I want to move the lookup of the catalog URL to the appcatalog repo but I'll do that as a later change.

We shouldn't use apptest to create the app CRs. It's only intended for use in tests. We use the CtrlClient so we don't need to import apiextensions which gets updated constantly and would create a lot of dependabot noise.